webapp/src/js/services/contacts.js

```diff
@@ -5,50 +5,57 @@ angular.module('inboxServices').factory('Contacts',
   function(
     $q,
     Cache,
-    ContactSchema,
+    ContactTypes,
     DB
   ) {
 
     'use strict';
     'ngInject';
 
-    var cacheByType = {};
-    ContactSchema.getPlaceTypes().forEach(function(type) {
-      cacheByType[type] = Cache({
-        get: function(callback) {
-          DB().query('medic-client/contacts_by_type', { include_docs: true, key: [type] })
-            .then(function(result) {
-              callback(null, _.pluck(result.rows, 'doc'));
-            })
-            .catch(callback);
-        },
-        invalidate: function(doc) {
-          return doc.type === type;
-        }
+    const init = ContactTypes.getPlaceTypes().then(types => {
+      const cacheByType = {};
+      types.forEach(type => {
+        cacheByType[type.id] = Cache({
+          get: function(callback) {
+            DB().query('medic-client/contacts_by_type', { include_docs: true, key: [type.id] })
+              .then(function(result) {
+                callback(null, _.pluck(result.rows, 'doc'));
+              })
+              .catch(callback);
+          },
+          invalidate: function(doc) {
+            return type.id === (doc.contact_type || doc.type);
```

Comment:
Maybe `(doc.type === 'contact' && doc.contact_type) || doc.type` ?

Reply:
I think this came up above too - this would be more defensive for other docs with a defined `contact_type` but I think that's unlikely.